File: py/minimumAddedCoins.py
class Solution:
    def minimumAddedCoins(self, coins: List[int], target: int) -> int:
        coins.sort()
        m = 1
        a = 0
        i = 0

        while m <= target:
            if i < len(coins) and coins[i] <= m:
                m += coins[i]
                i += 1
            else:
                m += m
                a += 1
        
        return a

File: cpp/minimumAddedCoins.cpp
#include <vector>
#include <algorithm>

using namespace std;

class Solution {
public:
    int minimumAddedCoins(vector<int>& coins, int target) {
        //[1,4,10,5,7,19], target = 19. X is obtainable if there is a subsequence of coins that adds to target.
        sort(coins.begin(), coins.end());
        
        int m = 1;
        int added = 0;
        int i = 0;

        while (m <= target) {
            if (i < coins.size() && coins[i] <= m) m += coins[i ++];
            else {
                m += m;
                added ++;
            }
        }
        return added;
    }
};